libraries/rush-lib/src/logic/operations/PhasedOperationPlugin.ts

```diff
@@ -71,10 +73,14 @@ function createOperations(
   function getOrCreateOperation(phase: IPhase, project: RushConfigurationProject): Operation {
     const key: string = getOperationKey(phase, project);
     let operation: Operation | undefined = operations.get(key);
+    const operationSettings: IOperationSettings | undefined = projectConfigurations
+      .get(project)
+      ?.operationSettingsByOperationName.get(phase.name);
```

Comment:
Move this inside of the `if`, since we don't need to refetch when returning from cache.